unstructured/ingest/connector/pinecone.py

```diff
@@ -87,26 +87,28 @@ def upsert_batch(self, batch):
             raise WriteError(f"http error: {api_error}") from api_error
         logger.debug(f"results: {response}")
 
-    def write_dict(self, *args, dict_list: t.List[t.Dict[str, t.Any]], **kwargs) -> None:
+    def write_dict(self, *args, elements_dict: t.List[t.Dict[str, t.Any]], **kwargs) -> None:
         logger.info(
-            f"Upserting {len(dict_list)} elements to destination "
+            f"Upserting {len(elements_dict)} elements to destination "
             f"index at {self.connector_config.index_name}",
         )
 
         pinecone_batch_size = self.write_config.batch_size
 
         logger.info(f"using {self.write_config.num_processes} processes to upload")
         if self.write_config.num_processes == 1:
-            for chunk in chunk_generator(dict_list, pinecone_batch_size):
+            for chunk in chunk_generator(elements_dict, pinecone_batch_size):
                 self.upsert_batch(chunk)  # noqa: E203
 
         else:
             with mp.Pool(
                 processes=self.write_config.num_processes,
             ) as pool:
-                pool.map(self.upsert_batch, list(chunk_generator(dict_list, pinecone_batch_size)))
+                pool.map(
+                    self.upsert_batch, list(chunk_generator(elements_dict, pinecone_batch_size))
+                )
 
-    def write(self, docs: t.List[BaseIngestDoc]) -> None:
+    def get_elements_dict(self, docs: t.List[BaseSingleIngestDoc]) -> t.List[t.Dict[str, t.Any]]:
         dict_list: t.List[t.Dict[str, t.Any]] = []
         for doc in docs:
             local_path = doc._output_filename
```

Comment:
Lines 119-138 are still not being called by `write_elements`, and they are essential while writing content. We need to isolate those parts as well. Both `write` and `write_elements` should work